includes/definitions/discovery/digipower.yaml

```diff
@@ -1,4 +1,41 @@
+mib: DigiPower-PDU-MIB
 modules:
   os:
-    hardware: .1.3.6.1.4.1.17420.1.2.9.1.19.0
-    version: .1.3.6.1.4.1.17420.1.2.4.0
+    hardware: DigiPower-PDU-MIB::pdu01ModelNo.0
+    version: DigiPower-PDU-MIB::devVersion.0
+    serial: DigiPower-PDU-MIB::devMAC.0
+  sensors:
+    current:
+      options:
+        divisor: 10
+        skip_value_lt: 0
+      data:
+        - oid: 'pduTable'
+          value: 'pdu01Value' 
+          num_oid: '.1.3.6.1.4.1.17420.1.2.9.1.11.{{ $index }}'
+          index: 0
+          descr: 'Current'
+          warn_limit: 'pdu01Threshold1'
+          high_limit: 'pdu01Threshold2'
+    voltage:
+      options:
+        skip_value_lt: 0
+      data:
+        - oid: 'pduTable'
+          value: 'pdu01Voltage'
+          num_oid: '.1.3.6.1.4.1.17420.1.2.9.1.17.{{ $index }}'
+          descr: 'Voltage'
+          index: 0
+    power:
+      options:
+        divisor: 10
+        skip_value_lt: 0
+      data:
+        - oid: 'pduTable'
+          value: 'pdu01Value' 
+          num_oid: '.1.3.6.1.4.1.17420.1.2.9.1.11.{{ $index }}'
+          index: 0
+          descr: 'Power'
+          multiplier: 'pdu01Voltage'
```

Comment:
Hello @gunkaaa 
multiplier must be a constant unfortunately. So this is currently not possible to compute the power out of the current and voltage. It can be done in PHP if you like.

Reply:
it works, and apparently intentionally so - `YamlDiscovery::replaceValues` is deliberately called for divisor and multiplier, using the exact same codepath as for `['low_limit', 'low_warn_limit', 'warn_limit', 'high_limit']` (where precache/walked table replacements are used across dozens of definitions)

https://github.com/librenms/librenms/blob/4561bf8d8864f4cc202f77bd178932bd344b8536/includes/discovery/functions.inc.php#L717 

https://github.com/librenms/librenms/blob/4561bf8d8864f4cc202f77bd178932bd344b8536/LibreNMS/Device/YamlDiscovery.php#L182-L184

https://github.com/librenms/librenms/blob/4561bf8d8864f4cc202f77bd178932bd344b8536/LibreNMS/Device/YamlDiscovery.php#L241-L248

I've added debug logging for this in 94891c522 -

with `oid`, `value`, `multiplier` all fully qualified: 

```
Sensor fetched value: 5  
Using 230 from discovery data for DigiPower-PDU-MIB::pdu01Voltage  
Setting value of multiplier to 230  
Discovered Sensor: .1.3.6.1.4.1.17420.1.2.9.1.11.0, 0, digipower, Power, snmp, 10, 230, , 115, (limits: LL: , LW: , W: , H: ), rrd_type = GAUGE  
SQL[update `sensors` set `sensor_current` = ? where `sensor_id` = ? [115,245] 1.87ms] 
```

with `oid`, `value`, `multiplier` not qualified:
```
Sensor fetched value: 6  
Using 230 from discovery data for pdu01Voltage  
Setting value of multiplier to 230  
Discovered Sensor: .1.3.6.1.4.1.17420.1.2.9.1.11.0, 0, digipower, Power, snmp, 10, 230, , 138, (limits: LL: , LW: , W: , H: ), rrd_type = GAUGE  
```

for the case where OID qualification doesn't match up I've added warnings in 713e0663f :

```
  'data' => 
  array (
    0 => 
    array (
      'oid' => 'DigiPower-PDU-MIB::pduTable',
      'value' => 'DigiPower-PDU-MIB::pdu01Value',
      'num_oid' => '.1.3.6.1.4.1.17420.1.2.9.1.11.{{ $index }}',
      'index' => 0,
      'descr' => 'Power',
      'multiplier' => 'pdu01Voltage',
    ),
  ),
(...)
Sensor fetched value: 5  
Valid number not found in data for multiplier, defaulting to 1  
Discovered Sensor: .1.3.6.1.4.1.17420.1.2.9.1.11.0, 0, digipower, Power, snmp, 10, 1, , 0.5, (limits: LL: , LW: , W: , H: ), rrd_type = GAUGE  
```